File: JavaScriptSolutions/_0007Solution.js
// https://leetcode.com/problems/reverse-integer/
// medium
// 2119, 2442

/**
 * @param {number} x
 * @return {number}
 */
var reverse1 = function (x) {
  let result = 0;

  let idx = x < 0;
  x = Math.abs(x);
  while (x > 0) {
      result = result * 10 + x % 10;
      x = Math.floor(x / 10);
  }

  return result > 0x7FFFFFFF ? 0 : idx ? -result : result;
};

var reverse = function (x) {
  let idx = x < 0;
  let y = "" + Math.abs(x);
  let result = +(y.split("").reverse().join(""));

  return result > 0x7FFFFFFF ? 0 : idx ? -result : result;

};

File: JavaSolutions/src/_0007Solution.java
// https://leetcode.com/problems/reverse-integer/
// medium
// Time: O(log x); Space: O(1)
// 2119, 2442

public class _0007Solution {

  public static void main(String[] args) {
    int i1 = 123;
    int i2 = -123;
    int i3 = 120;

    System.out.println(reverse(i1));
    System.out.println(reverse(i2));
    System.out.println(reverse(i3));

  }

  public static int reverse(int x) {
    long result = 0;

    while(x != 0) {
      result = result * 10 + x % 10;
      x /= 10;
    }

    return (result < Integer.MIN_VALUE || result > Integer.MAX_VALUE) ? 0 : (int) result;
  }
}
